Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -230,19 +268,44 @@ function addAttributeToRenderResources(
     renderResources.featureIdVertexAttributeSetIndex = setIndex + 1;
   }
 
+  // The position attribute should always be in the first index.
+  const isPositionAttribute = semantic === VertexAttributeSemantic.POSITION;
+  const index = isPositionAttribute ? 0 : attributeIndex;
```

Comment:
I'm a little confused, from code earlier in this file, the 2D position has a semantic of POSITION right? is that returning 0 in this case?

Reply:
2D position will always use `attributeIndex`. This just allows us to set the 3D position with an index of 0, and the 2D position as the next attribute index